base/base_tests/range_iterator_test.cpp

```diff
@@ -0,0 +1,43 @@
+#include "testing/testing.hpp"
+
+#include "base/range_iterator.hpp"
+
+#include "std/vector.hpp"
+
+UNIT_TEST(RangeIterator)
+{
+  using namespace my;
+
+  {
+    vector<int> result;
+    for (auto const i : Range(5))
+      result.push_back(i);
+    TEST_EQUAL(result, (vector<int>{0, 1, 2, 3, 4}), ());
+  }
+
+  {
+    vector<int> result;
+    for (auto const i : Range(2, 5))
+      result.push_back(i);
+    TEST_EQUAL(result, (vector<int>{2, 3, 4}), ());
+  }
+
+  {
+    vector<int> result;
+    for (auto const i : ReverseRange(5))
+      result.push_back(i);
+    TEST_EQUAL(result, (vector<int>{4, 3, 2, 1, 0}), ());
+  }
+
+  {
+    vector<int> result;
+    for (auto const i : ReverseRange(2, 5))
```

Comment:
IMHO the naming is confusing, especially if we're going to compare ReverseRange with haskell list generators or lisp loops. How about to rename ReverseRange to DownTo(from, to), where DownTo generates a list [from, from - 1, ..., to], not [to, to - 1, ..., from]. Also, Range can be renamed to UpTo(). @deathbaba @mgsergio WDYT?